File: src/CPP/hello.cc
#include <iostream>
#include <string>

using namespace std;

int main(int argc, char *argv[]) {
//	If there is no arguments ask for name
	if (argc == 1) {
		cout << "Hello, what is your name? > ";

		string name;
		cin >> name;
		cout << "Hello, " << name << endl;
	} else {
//  Else print argument
		cout << "Hello, " << argv[1] << endl;
	}
}

File: src/CSharp/hello.cs
using System;

class Hello {
    public static int Main(string[] args) {
        if (args.Length < 1) { 
            Console.WriteLine("./{name of the executable} {Name}"); return 1;
        }
        Console.WriteLine("Hello, World!");
        Console.WriteLine("Hello, " + args[0]);
        return 0;
    }
}
